packages/mui-system/src/cssVars/createCssVarsProvider.js

```diff
@@ -147,12 +177,34 @@ export default function createCssVarsProvider(options) {
         shouldSkipGeneratingVar,
       });
       theme.vars = deepmerge(theme.vars, vars);
+
       if (key === calculatedColorScheme) {
+        const restParsedScheme = { ...parsedScheme };
+
+        if (additionalColorTokensPaths.length > 0) {
```

Comment:
Instead of creating another parameter, can we just shallow merge the color scheme props to the theme?

Meaning:
```js
// before calculating the color scheme
theme = {
  ...,
  ref: { typeface: { ... } },
  sys: { typeface: { ... } },
}

// after the color scheme is calculated
theme = {
  ...,
  ref: { typeface: { ... }, palette: { ... } },
  sys: { typeface: { ... }, color: { ... } },
}
```

Reply:
Would this be better? This removes the need for another configuration and should work with most cases.

```diff
diff --git a/packages/mui-system/src/cssVars/createCssVarsProvider.js b/packages/mui-system/src/cssVars/createCssVarsProvider.js
index 933e0e9079..499eedf69f 100644
--- a/packages/mui-system/src/cssVars/createCssVarsProvider.js
+++ b/packages/mui-system/src/cssVars/createCssVarsProvider.js
@@ -149,7 +149,17 @@ export default function createCssVarsProvider(options) {
       theme.vars = deepmerge(theme.vars, vars);
       if (key === calculatedColorScheme) {
         // 4.1 Merge the selected color scheme to the theme
-        theme = { ...theme, ...parsedScheme };
+        Object.keys(parsedScheme).forEach((schemeKey) => {
+          if (parsedScheme[schemeKey] && typeof parsedScheme[schemeKey] === 'object') {
+            // shallow merge the 1st level structure of the theme.
+            theme[schemeKey] = {
+              ...theme[schemeKey],
+              ...parsedScheme[schemeKey],
+            };
+          } else {
+            theme[schemeKey] = parsedScheme[schemeKey];
+          }
+        });
         if (theme.palette) {
           theme.palette.colorScheme = key;
         }
```

I don't think there will be a case like this:
```js
extendTheme({
  ref: {
    typography: { ... },
    color: { ... }, // We can neglect this case because the color-related tokens should go to color schemes.
  },
  colorSchemes: {
    light: {
      ref: {
        color: { ... },
      }
    },
    dark: {
      ref: {
        color: { ... },
      }
    }
  }
})
```

Reply:
Yep, seems like this works, I push the change.